contracts/Comptroller/Diamond/facets/MarketFacet.sol

```diff
@@ -33,7 +33,25 @@ contract MarketFacet is IMarketFacet, FacetBase {
      * @return A dynamic list with the assets the account has entered
      */
     function getAssetsIn(address account) external view returns (VToken[] memory) {
-        return accountAssets[account];
+        uint256 len = 0;
+        for (uint256 i = 0; i < accountAssets[account].length; i++) {
+            Market storage market = markets[address(accountAssets[account][i])];
+            if (market.isListed) {
+                len++;
+            }
+        }
+
+        VToken[] memory assetsIn = new VToken[](len);
+        uint256 index = 0;
+        for (uint256 i = 0; i < accountAssets[account].length; i++) {
+            Market storage market = markets[address(accountAssets[account][i])];
+            if (market.isListed) {
+                assetsIn[index] = accountAssets[account][i];
+                index++;
+            }
+        }
+
+        return assetsIn;
     }
```

Comment:
I think we can optimise this piece of code as below

function getAssetsIn(address account) external view returns (VToken[] memory) {

        uint256 len;
        VToken[] storage accountAssetsArray = accountAssets[account];
        uint256 accountAssetsArrayLength = accountAssetsArray.length;

        VToken[] memory assetsIn = new VToken[](accountAssetsArrayLength);

        for (uint256 i; i < accountAssetsArrayLength; ++i) {
            Market storage market = markets[address(accountAssetsArray[i])];
            if (market.isListed) {
                assetsIn[len] = accountAssetsArray[i];
                ++len;
            }
        }

        assembly {
            mstore(assetsIn, len)
        }

        return assetsIn;
}